cli/commands/pods.go

```diff
@@ -24,29 +23,44 @@ func (cmd *podHandler) handleList(a *kingpin.Application, e *kingpin.ParseElemen
 }
 
 func (cmd *podHandler) handleStatus(a *kingpin.Application, e *kingpin.ParseElement, c *kingpin.ParseContext) error {
-	endpointPath := "v1/pod/status"
+	var endpointPath string
 	if len(cmd.PodName) > 0 {
 		endpointPath = fmt.Sprintf("v1/pod/%s/status", cmd.PodName)
+	} else {
+		endpointPath = "v1/pod/status"
 	}
 	body, err := client.HTTPServiceGet(endpointPath)
 	if err != nil {
 		client.PrintMessageAndExit(err.Error())
 	}
-	client.PrintJSONBytes(body)
+	if cmd.RawJSON {
+		client.PrintJSONBytes(body)
+	} else {
+		if len(cmd.PodName) > 0 {
+			client.PrintMessage(toPodTree(body))
+		} else {
+			client.PrintMessage(toServiceTree(body))
+		}
+	}
 	return nil
 }
 
-func toPodTree(podsJSONBytes []byte) string {
-	podsJSON, err := client.UnmarshalJSON(podsJSONBytes)
+func toServiceTree(podsJSONBytes []byte) string {
+	response, err := client.UnmarshalJSON(podsJSONBytes)
 	if err != nil {
 		client.PrintMessageAndExit(fmt.Sprintf("Failed to parse JSON in pods response: %s", err))
 	}
+
 	var buf bytes.Buffer
-	buf.WriteString(fmt.Sprintf("%s\n", config.ServiceName))
+	buf.WriteString(fmt.Sprintf("%s\n", response["service"]))
 
 	i := 0
-	for podName, rawPod := range podsJSON {
-		appendPod(&buf, podName, rawPod, i == len(podsJSON)-1)
+	rawPods, ok := response["pods"].([]interface{})
+	if !ok {
+		client.PrintMessageAndExit("Failed to parse JSON pods in response")
+	}
+	for _, rawPod := range rawPods {
```

Comment:
Why not replace the `_` with `i` locally? Then there would be no need for the post-increment.

Reply:
It is also not 100% on first reading that we are calculating `isLast` as the final parameter.

Reply:
Good point, fixed here and elsewhere